.travis.yml

```diff
@@ -1,42 +1,94 @@
 sudo: required
+_aliases:
+  - &ui-tests
```

Comment:
I get that this is a manual declaration of what we had in the matrix but it adds a lot of noise which will make it really hard to maintain. I suggest some simplifications:

- Use `node-prod` and `node-next` so we can update the version numbers easily
- Use more aliases to clean up duplicate code
- Preserve the old order of stage execution
- Remove the old `$TASK` approach which was only needed for the matrix
- Only run crucial jobs in `node-prod` and `node-next`, not all jobs. Crucial jobs are the build step and CI tests.

Here's my attempt at it. I haven't tried a build with this file though:

<details>
<summary>Show contents

`.travis.yml`

</summary>

```yaml
sudo: required
_aliases:
  - &tox
    language: python
    python: 3.6
    install: pip install tox
    script: tox
  - &node-prod
    # This is the NodeJS version we run in production.
    language: node_js
    node_js: '6'
    before_install: npm install -g yarn
  - &node-next
    # This is the next NodeJS version we will support.
    language: node_js
    node_js: '8'
    before_install: npm install -g yarn
services:
  - docker
cache:
  yarn: true
  directories:
  - node_modules
jobs:
  include:

    # Test the build process.
    - stage:
      <<: *node-prod
      env: ADDONS_FRONTEND_BUILD_ALL=1
      script: yarn build
    - stage:
      <<: *node-next
      env: ADDONS_FRONTEND_BUILD_ALL=1
      script: yarn build

    # Run the unit/integration tests.
    - stage:
      <<: *node-prod
      script: yarn test-ci
    - stage:
      <<: *node-next
      script: yarn test-ci

    - stage:
      <<: *node-prod
      script: yarn lint

    - stage:
      <<: *node-prod
      script: yarn nsp-check

    # Run the functional tests.
    - stage:
      <<: *tox
      <<: *node-prod
      addons:
        firefox: latest-nightly
        hosts: example.com
      env: TOXENV=discopane-ui-tests DISPLAY=:99.0 GECKODRIVER=0.19.1 MOZ_HEADLESS=1 PATH=./node_modules/.bin:$PATH
      install:
        - yarn
        - yarn start-func-test-server &
      before_script:
        - sudo apt-get -qq update
        - sudo apt-get install jwm
        - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GECKODRIVER/geckodriver-v$GECKODRIVER-linux64.tar.gz
        - mkdir $HOME/geckodriver && tar xvf /tmp/geckodriver.tar.gz -C $HOME/geckodriver
        - export PATH=$HOME/geckodriver:$PATH
        - firefox --version
        - geckodriver --version
        - pip install tox
        - jwm &
        - sleep 3
        # Wait for server to be available
        - until $(curl --output /dev/null --silent --head --fail -k https://example.com:4000); do printf '.'; sleep 5; done

    - stage:
      <<: *tox
      env: TOXENV=flake8

    - stage:
      <<: *tox
      env: TOXENV=dennis-lint
```

</details>